### ISSUE TITLE
[Task] CRC extension: Installation and setup (Windows)

### ISSUE THREAD

**Issue by gbraad** [containers/podman-desktop#1439]
The CRC extension needs to be able to install and setup the CRC tool, and enable the necessary virtualization; such as Hyper-V on Windows.

### Tasks
 * [x] Detect the existence of CRC
   * `crc` in path
 * [x] Optional: Is virtualization functional?
   * This is a necessity for a 'possible reboot' on Windows
   * Preflight can detect this (after install)
 * [x] Run the installation process
   * [x] Optional: [suppress the instalation of the tray](https://github.com/containers/podman-desktop-internal/issues/88) on Windows

### Expectation
After installation the CRC environment should be working. This might need a reboot, or installs the old tray. We can move functionality, like preflight checks around later, which might give an overview of the failures and fixes it can perform. For this initial version we assume that the MSI handles this setup. The issue we will have is to detect the need for a reboot.

### Notes
 * macOS package installs the hypervisor drivers.
 * Linux needs KVM to be available.
 * Windows needs Hyper-V, which requires a user to be added to a group and optional features to be enabled. It requires a reboot.

### Dependencies
 * Optional: https://github.com/crc-org/crc/issues/3507

**Comment by evidolob** [comment]
I propose to follow podman ext installation flow, so each OS build has own, build in,  installation package, we provide Windows and MacOS install first.

We always install latest version of CRC, by fetching them form `https://developers.redhat.com/content-gateway/rest/mirror/pub/openshift-v4/clients/crc/latest/release-info.json`

>The CRC extension needs to be able to install and setup the CRC tool, and enable the necessary virtualization; such as Hyper-V on Windows.

I not clear on this ` and enable the necessary virtualization;` 
Should CRC ext enable Hyper-V? That will require privileges escalations, and dealing with Win UAC.

Also, should we provide preflight checks for OS/hardware that user has?(like we do for podman)

**Comment by gbraad** [comment]
> Should CRC ext enable Hyper-V? That will require privileges escalations, and dealing with Win UAC.

At the moment I can not answer. The MSI can install the virtualization stack. Tis goes all the way back to: https://gist.github.com/gbraad/ab07131c0054abf2a0d5aa729454eb9c

If the MSI can be executed and take care of this, that would suffice for now. However, I am not sure if this will escalate privileges correctly if the installation is performed as `/quiet`. So this might have to be involving the actual UI of the MSI? WDYT?

> Also, should we provide preflight checks for OS/hardware that user has?(like we do for podman)

At the moment an open question. The preflights we normally perform from the `crc` executable. If you look at the epic there is a task called "Track progress of installation". In that task we will take the output of the setup step to follow progress... and this might aso include the preflights: looking something like: https://github.com/crc-org/tray-electron/issues/114#issue-1114579745 and https://github.com/crc-org/tray-electron/issues/114#issuecomment-1429653949. So not urgent now.

In short; if we can check if Hyper-V is available in some form, we are OK for now. Otherwise postpone.

**Comment by gbraad** [comment]
@evidolob should we break it down into platform related issues too? This might be helpful, as we do not really have a packaged flow for linux (especially as we only support a few distros out of the box).

**Comment by gbraad** [comment]
@evidolob I'll create the breakdown in the extension repo.

**Comment by gbraad** [comment]
Implemented for both macOS and Windows.
Minor changes, like tracking of the installation for progress is handled in follow-up issues.
